src/core/openthread.cpp

```diff
@@ -691,6 +691,13 @@ void otPlatformReset(otInstance *aInstance)
     otPlatReset(aInstance);
 }
 
+void otFactoryReset(otInstance *aInstance)
+{
+    aInstance->mThreadNetif.GetActiveDataset().Clear(true);
+    aInstance->mThreadNetif.GetPendingDataset().Clear(true);
```

Comment:
Rather than calling `Clear()` on the respective data objects, should this just call `otPlatSettingsWipe()` followed by `otPlatReset()`?